server/src/main/java/com/google/coffeehouse/servlets/GetProfileServlet.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to get a {@link Person} from Http GET Request Body (in JSON format)
+ * that exists in the database through a call to the Storage Handler API,
+ * and return it in JSON format.
+ */
+@WebServlet("/api/get-profile")
+public class GetProfileServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doGet when the body of the 
+   * GET request cannot be used to fetch a {@link Person} for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
```

Comment:
Consider having more specific error codes. For example:
- Unable to parse body (when parsing fails)
- Invalid argument (when it's missing a required field)
- Not found (when the profile being requested is not in the DB)

Reply:
When the profile is not found, the storage handler call itself returns an exception with the error message "This person does not exist in the database." This would mean that I wouldn't need create an exception error code for it within this file, unless of you think that would be better instead?

Reply:
Sorry, to clarify, my concern is less with this specific error string and more with the codes being returned. The only returned code appears to be SC_BAD_REQUEST, and a few others would probably be appropriate.

Reply:
Oh that makes sense, thank you for the clarification!